cli/lock.ml

```diff
@@ -189,25 +192,72 @@ let display_verbose_diagnostics = function
   | None -> false
   | Some l -> l >= Logs.Info
 
-let interpret_solver_error ~switch_state = function
+let interpret_solver_error ~repositories solver = function
   | `Msg _ as err -> err
   | `Diagnostics d ->
-      (match Opam_solve.not_buildable_with_dune d with
+      (match Opam_solve.not_buildable_with_dune solver d with
       | [] -> ()
       | offending_packages ->
-          check_dune_universe_repo ~switch_state offending_packages);
+          check_dune_universe_repo ~repositories offending_packages);
       let verbose = display_verbose_diagnostics (Logs.level ()) in
-      Opam_solve.diagnostics_message ~verbose d
+      Opam_solve.diagnostics_message ~verbose solver d
 
-let calculate_opam ~build_only ~allow_jbuilder ~local_opam_files ~ocaml_version
-    ~target_packages =
+let pull_repository url =
```

Comment:
That's because it only supports local FS URLs for now, for remote ones it will pull them but you're right it could be better named. The idea is that it takes a list of URLs and make sure each of the corresponding repos is available in a local folder so something like `repository_url_to_local_folder` would make it totally unambiguous.

Reply:
Ah I see. Yeah, that could be a better name. Or `url_to_local_path`.

Reply:
I'm just afraid that once we had the pulling here, it might be unclear that it actually happens in that function. We can improve it then!

Reply:
Also note that we already have a similar `pull_pin_depends` function that does pretty much the same thing!